src/libxrpl/protocol/STIssue.cpp

```diff
@@ -40,23 +40,46 @@ STIssue::STIssue(SField const& name) : STBase{name}
 
 STIssue::STIssue(SerialIter& sit, SField const& name) : STBase{name}
 {
-    issue_.currency = sit.get160();
-    if (!isXRP(issue_.currency))
-        issue_.account = sit.get160();
-    else
-        issue_.account = xrpAccount();
-
-    if (isXRP(issue_.currency) != isXRP(issue_.account))
-        Throw<std::runtime_error>(
-            "invalid issue: currency and account native mismatch");
-}
+    auto const currencyOrAccount = sit.get160();
 
-STIssue::STIssue(SField const& name, Issue const& issue)
-    : STBase{name}, issue_{issue}
-{
-    if (isXRP(issue_.currency) != isXRP(issue_.account))
-        Throw<std::runtime_error>(
-            "invalid issue: currency and account native mismatch");
+    if (isXRP(static_cast<Currency>(currencyOrAccount)))
+    {
+        asset_ = xrpIssue();
+    }
+    // Check if MPT
+    else
+    {
+        // MPT is serialized as:
+        // - 160 bits MPT issuer account
+        // - 160 bits black hole account
+        // - 32 bits sequence
+        AccountID account = static_cast<AccountID>(sit.get160());
+        // MPT
+        if (noAccount() == account)
+        {
+            MPTID mptID;
+            std::uint32_t sequence = sit.get32();
+            static_assert(
+                MPTID::size() == sizeof(sequence) + sizeof(currencyOrAccount));
+            memcpy(mptID.data(), &sequence, sizeof(sequence));
+            memcpy(
+                mptID.data() + sizeof(sequence),
+                currencyOrAccount.data(),
+                sizeof(currencyOrAccount));
+            MPTIssue issue{mptID};
+            asset_ = issue;
+        }
+        else
+        {
+            Issue issue;
+            issue.currency = currencyOrAccount;
+            issue.account = account;
+            if (!isConsistent(issue))
+                Throw<std::runtime_error>(
```

Comment:
can you include this line in unit tests (it would require non-XRP currency, XRP account)